genericjpa/common-apis/src/main/java/org/hibernate/search/genericjpa/metadata/impl/MetadataRehasher.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.genericjpa.metadata.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.hibernate.search.backend.spi.SingularTermDeletionQuery.Type;
+import org.hibernate.search.engine.metadata.impl.DocumentFieldMetadata;
+import org.hibernate.search.engine.metadata.impl.EmbeddedTypeMetadata;
+import org.hibernate.search.engine.metadata.impl.PropertyMetadata;
+import org.hibernate.search.engine.metadata.impl.TypeMetadata;
+import org.hibernate.search.exception.AssertionFailure;
+import org.hibernate.search.metadata.NumericFieldSettingsDescriptor.NumericEncodingType;
+
+/**
+ * @author Martin Braun
+ */
+public final class MetadataRehasher {
+
+	public List<RehashedTypeMetadata> rehash(List<TypeMetadata> originals) {
+		return originals.stream().map( this::rehash ).collect( Collectors.toList() );
```

Comment:
This is using Java 8. Is that really what we want? As said before I'd vote against it if there is no real advantage from the user's PoV. So I'd like to make sure we all agree on this move.

/CC @Sanne.